plugins/services/src/js/containers/service-detail/ServiceDetail.js

```diff
@@ -182,59 +154,78 @@ class ServiceDetail extends mixin(TabsMixin) {
   getTabs() {
     const {service:{id}} = this.props;
     const routePrefix = `/services/overview/${encodeURIComponent(id)}`;
+    const routeDefinitionPrefix = '/services/overview/:id';
 
     const tabs = [];
-    const activeTab = this.state.currentTab;
+    const {currentTab} = this.state;
 
     tabs.push({
       label: 'Instances',
       callback: () => {
-        this.setState({currentTab: 'tasks'});
+        this.context.router.push(`${routePrefix}/tasks`);
       },
-      isActive: activeTab === 'tasks'
+      isActive: currentTab === `${routeDefinitionPrefix}/tasks`
     });
 
     tabs.push({
       label: 'Configuration',
       callback: () => {
-        this.setState({currentTab: 'configuration'});
+        this.context.router.push(`${routePrefix}/configuration`);
       },
-      isActive: activeTab === 'configuration'
+      isActive: currentTab === `${routeDefinitionPrefix}/configuration`
     });
 
     tabs.push({
       label: 'Debug',
       callback: () => {
-        this.setState({currentTab: 'debug'});
+        this.context.router.push(`${routePrefix}/debug`);
       },
-      isActive: activeTab === 'debug'
+      isActive: currentTab === `${routeDefinitionPrefix}/debug`
     });
 
     if (this.hasVolumes()) {
       tabs.push({
-        label: 'Volumes', routePath: routePrefix + '/volumes',
+        label: 'Volumes',
         callback: () => {
-          this.setState({currentTab: 'volumes'});
+          this.context.router.push(`${routePrefix}/volumes`);
         },
-        isActive: activeTab === 'volumes'
+        isActive: currentTab === `${routeDefinitionPrefix}/volumes`
       });
     }
 
     return tabs;
   }
 
   render() {
-    const {children, service:{id}} = this.props;
-    const breadcrumbs = <ServiceBreadcrumbs serviceID={id} />;
+    const {children, actions, errors, params, routes, service} = this.props;
+    const breadcrumbs = <ServiceBreadcrumbs serviceID={service.id} />;
+    const clonedProps = {
+      params,
+      routes,
+      service,
+      errors: MarathonErrorUtil.parseErrors(errors[ActionKeys.SERVICE_EDIT]),
+      onClearError: this.handleEditClearError,
+      onEditClick: actions.editService,
+      volumes: service.getVolumes().getItems()
+    };
+
+    const clonedChildren = React.Children.map(children, function (child) {
+      // Only add props to children that are not ServiceModals
+      if (child != null && child.type === ServiceModals) {
+        return child;
+      }
+
+      return React.cloneElement(child, clonedProps);
```

Comment:
We need to pass down the props that was passed down directly in the `renderxxxTab` functions